site/src/lib/model/apiKey.model.ts

```diff
@@ -0,0 +1,222 @@
+import crypto from "crypto";
+import { Db, DBRef } from "mongodb";
+import { v4 as uuid } from "uuid";
+import { mustGetEnvVar } from "../../util/api";
+
+import { User } from "./user.model";
+
+type ApiKeyProperties = {
+  createdAt: Date;
+  displayName: string;
+  hashedString: string;
+  lastUsedAt?: Date | null;
+  publicId: string;
+  revokedAt?: Date | null;
+  salt: string;
+  useCount: number;
+  user: DBRef;
+};
+
+export type UserFacingApiKeyProperties = Omit<
+  ApiKeyProperties,
+  "hashedString" | "salt" | "user"
+>;
+
+type ApiKeyDocument = ApiKeyProperties;
+
+export class ApiKey {
+  createdAt: Date;
+  displayName: string;
+  hashedString: string;
+  lastUsedAt?: Date | null;
+  publicId: string;
+  revokedAt?: Date | null;
+  salt: string;
+  useCount: number;
+  user: DBRef;
+
+  static readonly COLLECTION_NAME = "bp-api-keys";
+
+  private static readonly HASHING_SECRET = mustGetEnvVar("HASHING_SECRET");
+
+  /**
+   * This prefix helps to identify keys for both humans and automated code-scanners.
+   * It aims to balance human-readability with avoiding false positives (which plain "blocks" might do).
+   */
+  private static readonly PREFIX = "b10ck5";
+
+  private static readonly KEY_FORMAT_REGEXP = new RegExp(
+    // e.g. b10ck5.e84a616cedc8822b02ac96761b068f19.91ea0c90-acee-4215-a01e-85a4438af0ea
+    `^${ApiKey.PREFIX}\\.[a-z0-9]{32}\\.[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}$`,
```

Comment:
Non-blocking - could it be helpful to extract the publicId length `{32}` into a const that the generated bytes can use below?

Reply:
Agreed, thank you - I've done this [here](https://github.com/blockprotocol/blockprotocol/pull/108/commits/1c004c374c9935397a3969731a46a4bd9c9109e7)